lib/rucio/core/rse_expression_parser.py

```diff
@@ -91,7 +92,13 @@ def parse_expression(expression, filter=None, session=None):
         result = []
         for rse in list(result_tuple[0]):
             result.append(result_tuple[1][rse])
-        REGION.set(sha256(expression.encode()).hexdigest(), result)
+        # Don't cache the result if we only specified vo
+        vo_pattern = r'vo=[a-zA-Z0-9]+'
```

Comment:
Improvement: `$` matches the end of the string, so `r'vo=[a-zA-Z0-9]+$'` will match strings that contain only the VO and nothing else. This avoids the need for the `elif` statement.